lib/util/common.ts

```diff
@@ -11,14 +11,19 @@ import { parseUrl } from './url';
  * @param url the url to detect `platform` from
  * @returns matched `platform` if found, otherwise `null`
  */
-export function detectPlatform(url: string): 'gitlab' | 'github' | null {
+export function detectPlatform(
+  url: string
+): 'gitlab' | 'github' | 'azure' | null {
   const { hostname } = parseUrl(url) ?? {};
   if (hostname === 'github.com' || hostname?.includes('github')) {
     return 'github';
   }
   if (hostname === 'gitlab.com' || hostname?.includes('gitlab')) {
     return 'gitlab';
   }
+  if (hostname === 'dev.azure.com') {
```

Comment:
`visualstudio.com` is also a valid hostname. `dev.azure.com/myorg` and `myorg.visualstudio.com` are equivalent.

Reply:
added support for `*.visualstudio.com`